infrastructure/code/ignores_test.go

```diff
@@ -64,7 +64,6 @@ func Test_IgnoresWithNegationInSnykCode(t *testing.T) {
 
 	calls := codeClientMock.GetAllCalls("extendBundleWithSource")
 	assert.Len(t, calls, 1)
-	assert.Contains(t, scanner.ignorePatterns, "!"+filepath.ToSlash(repobase+"/**/temp"))
```

Comment:
The test doesn't make sense without this assertion.

Reply:
@asaf92

Reply:
You're right, I'll refactor the test (and the one below it)

Reply:
I changed this and other ignore tests to be more generic, and test the call to the mock when scanning rather than the results of the private filtering functions